packages/network-controller/CHANGELOG.md

```diff
@@ -5,6 +5,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+## [15.2.1]
+### Fixed
+- Align `Provider` type to be compatible with `@metamask/eth-query` ([#2028](https://github.com/MetaMask/core/pull/2028))
```

Comment:
Actually, now that I'm re-reading this, my original PR title was misleading. The provider types were already correct, but internally we were using an older version of `eth-query` which wasn't aligned with the provider type. There were no functional changes to the API with this bump, it just made it so that we could some extraneous `// @ts-expect-error`s. The patch bump is fine, but I'm not sure that we need this entry, since consumers shouldn't have to do anything in response to this change. What do you think?

Reply:
Hmm it's ambiguous, but since the `Provider` type is being exported by `network-controller`, I think it makes sense to treat it as part of its interface and catalogue any updates to it in CHANGELOG.

The fact that the `Provider` type update actually happens entirely inside of `eth-query` seems like an implementation detail. Users are still interacting with a new interface for `network-controller`, and probably need to be informed that it will behave differently.

Reply:
> Even though the Provider type changes actually happens entirely inside of `eth-query`, users are still interacting with a new interface for `network-controller`

Hmm. Nothing changed about the provider interface (that's what I mean about my PR title being misleading). It's still the same as it was before.

That said, I suppose you could say that since `network-controller` exports EthQuery as the `getEthQuery` interface, _that_ interface has changed. I can see that breaking people's code.